src/pages/content.tsx

```diff
@@ -155,28 +155,16 @@ const Content = () => {
             className="flex h-full flex-col overflow-auto bg-gray-100 p-6"
             id="content-view"
         >
-            {selectedNode && selectedDocument ? (
+            {selectedDocument ? (
                 <div className="flex-1 rounded-2xl bg-gray-50 p-4">
-                    <Suspense
```

Comment:
This suspense shows the loading indicator while the editor component is being loaded. The editors exported from document-model-libs are lazy components so they are only loaded the first time a document is opened.

This is easier too see with the document model editor which is pretty heavy. Unless are dealing with this elsewhere now?

Maybe we could adapt this to also show the loading indicator when a different document type is opened?
We also want to show it when the user opens a document and it is not yet fully synced. This doesn't have to be done in this PR but might be good to keep in mind.

Reply:
ah I see — the logic as I had it essentially did do this by checking for the existence of the editor (i.e. it hasn't loaded yet) and also then showed it if a different document type is loading (editor document types does not include document). But I agree its better to also make sure we show the loader if the async component is not yet loaded via suspense.

I've added an update now that wraps the async editor component in a suspense and shows the same loader for that. I extracted the loader component so we can update it to handle the not fully synced case as well. I put this in the editor component so that the `Content` component only needs to worry about whether to show a document editor or a folder view.

What do you think?

Reply:
It's great!
Just had a feature idea, we could allow a loader component to be set for each editor, so they could make a skeleton UI to be shown while loading.